code/game/turfs/simulated/floor/fancy_floor.dm

```diff
@@ -32,20 +32,49 @@
 	icon_state = "grass"
 	floor_tile = /obj/item/stack/tile/grass
 	broken_states = list("sand")
+	var/ore_type = /obj/item/weapon/ore/glass
 
 /turf/open/floor/grass/New()
 	..()
 	spawn(1)
 		update_icon()
 
 /turf/open/floor/grass/attackby(obj/item/C, mob/user, params)
+	if(istype(C, /obj/item/weapon/shovel) && params)
+		new ore_type(src)
+		new ore_type(src) //Make some sand if you shovel grass
+		user << "<span class='notice'>You shovel [src].</span>"
+		make_plating()
 	if(..())
 		return
-	if(istype(C, /obj/item/weapon/shovel))
-		new /obj/item/weapon/ore/glass(src)
-		new /obj/item/weapon/ore/glass(src) //Make some sand if you shovel grass
-		user << "<span class='notice'>You shovel the grass.</span>"
-		make_plating()
+
+/turf/open/floor/grass/snow
+	name = "Snow"
+	icon = 'icons/turf/snow.dmi'
+	icon_state = "snow"
+	ore_type = /obj/item/stack/sheet/mineral/snow
+	planetary_atmos = TRUE
+	floor_tile = null
+	initial_gas_mix = "o2=22;n2=82;TEMP=180"
+
+
+/turf/open/floor/grass/snow/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/crowbar))//You need to dig this turf out instead of crowbarring it
+		return
+	..()
+
+/turf/open/floor/grass/snow/basalt //By your powers combined, I am captain planet
+	name = "volcanic floor"
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "basalt"
+	ore_type = /obj/item/weapon/ore/glass/basalt
+	initial_gas_mix = "o2=14;n2=23;TEMP=300"
+
+/turf/open/floor/grass/snow/basalt/New()
+	..()
+	var/randumb = rand(0,12)
```

Comment:
This should mimic actual basalt;

    if(prob(15))
    	icon_state = "basalt[rand(0, 12)]"
    	switch(icon_state)
		if("basalt1", "basalt2", "basalt3")
			SetLuminosity(1, 1)